File: src/main/java/org/jukeboxmc/block/behavior/BlockBeetroot.java
package org.jukeboxmc.block.behavior;

import org.cloudburstmc.nbt.NbtMap;
import org.jukeboxmc.block.data.BlockCrops;
import org.jukeboxmc.item.Item;
import org.jukeboxmc.item.ItemType;
import org.jukeboxmc.item.enchantment.EnchantmentType;
import org.jukeboxmc.util.Identifier;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.concurrent.ThreadLocalRandom;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockBeetroot extends BlockCrops {

    public BlockBeetroot( Identifier identifier ) {
        super( identifier );
    }

    public BlockBeetroot( Identifier identifier, NbtMap blockStates ) {
        super( identifier, blockStates );
    }

    @Override
    public List<Item> getDrops( Item item ) {
        if ( !this.isFullyGrown() ) {
            return Collections.singletonList( Item.create( ItemType.BEETROOT_SEEDS ) );
        }
        int amount = 1;
        int attempts = 3 + Math.min( 0, item.hasEnchantment( EnchantmentType.FORTUNE ) ? item.getEnchantment( EnchantmentType.FORTUNE ).getLevel() : 0 );
        ThreadLocalRandom random = ThreadLocalRandom.current();

        for ( int i = 0; i < attempts; i++ ) {
            if ( random.nextInt(7) < 4 ) {
                amount++;
            }
        }

        return Arrays.asList( Item.create( ItemType.BEETROOT ), Item.create( ItemType.BEETROOT_SEEDS ).setAmount( amount ) );
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockBeetroot.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.Beetroot
import org.jukeboxmc.api.item.Item
import org.jukeboxmc.api.item.ItemType
import org.jukeboxmc.api.item.enchantment.EnchantmentType
import org.jukeboxmc.server.block.JukeboxBlock
import kotlin.math.min
import kotlin.random.Random

class BlockBeetroot(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates), Beetroot {

    override fun canPassThrough(): Boolean {
        return true
    }

    override fun getGrowth(): Int {
        return this.getIntState("growth")
    }

    override fun setGrowth(value: Int): BlockBeetroot {
        return this.setState("growth", value)
    }

    override fun getDrops(item: Item): MutableList<Item> {
        return if (this.getGrowth() != 7) {
            mutableListOf(Item.create(ItemType.BEETROOT_SEEDS, 1))
        } else {
            var amount = 2
            val attempts = 3 + min(
                0, if (item.hasEnchantment(EnchantmentType.FORTUNE)) item.getEnchantment(
                    EnchantmentType.FORTUNE
                )!!.getLevel() else 0
            )
            for (i in 0..<attempts) {
                if (Random.nextDouble() <= 0.5714286) {
                    amount++
                }
            }
            mutableListOf(Item.create(ItemType.BEETROOT_SEEDS, amount), Item.create(ItemType.BEETROOT, 1))
        }
    }
}